builtin/provisioners/chef-client/ssh_provisioner.go

```diff
@@ -0,0 +1,72 @@
+package chefclient
+
+import (
+	"bytes"
+	"strings"
+
+	"github.com/hashicorp/terraform/communicator"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func (p *Provisioner) sshInstallChefClient(
+	o terraform.UIOutput,
+	comm communicator.Communicator) error {
+	var installCmd bytes.Buffer
+
+	// Build up a single command based on the given config options
+	installCmd.WriteString("curl")
+	if p.HTTPProxy != "" {
+		installCmd.WriteString(" --proxy " + p.HTTPProxy)
+	}
+	if p.NOProxy != nil {
+		installCmd.WriteString(" --noproxy " + strings.Join(p.NOProxy, ","))
+	}
+	installCmd.WriteString(" -LO https://www.chef.io/chef/install.sh 2>/dev/null &&")
+	if !p.PreventSudo {
+		installCmd.WriteString(" sudo")
+	}
+	installCmd.WriteString(" bash ./install.sh")
+	if p.Version != "" {
+		installCmd.WriteString(" -v " + p.Version)
+	}
+	installCmd.WriteString(" &&")
+	if !p.PreventSudo {
+		installCmd.WriteString(" sudo")
+	}
+	installCmd.WriteString(" rm -f install.sh")
```

Comment:
Well if the Chef installation fails the resource will be marked as tainted and will be recreated on the next TF run. And if it doesn't fail, the file will be deleted. So don't see any point in making sure that we delete the file in a separate step if something failed as the node as a whole will be recreated anyway :smiley: